### PR TITLE
Visitor CRUD

### PR DESCRIPTION
**Pull request by haydarai**
This PR creates these endpoints:
- `GET /visitor`, get all visitors.
- `GET /visitor/:id`, get specified visitor.
- `PUT /visitor/:id`, update specified visitor. Only available to update `otherFields` array.

Make sure you've logged in before trying it out on Postman or Insomnia, because the current auth use cookies for authentication.

### CODE REVIEW

File: controllers/visitor.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    res.json({ visitor });
+    // todo: render visitor detail page
+  });
+};
+
+/**
+ * POST /visitor
+ * Create a new visitor data 
+ */
+exports.postVisitor = (req, res, next) => {
```
Comment (dwiajik) [inline controllers/visitor.js]: This does not have any authorization yet. Is this expected?

File: models/Visitor.js [hunk 2]
```diff
@@ -13,6 +14,7 @@ const visitorSchema = new mongoose.Schema({
     value: String
   }]
 }, { timestamps: true });
+visitorSchema.plugin(paginate);
```
Comment (dwiajik) [inline models/Visitor.js]: Just a nitpick, I think airbnb style suggest to have blank lines after block. https://github.com/airbnb/javascript#whitespace--after-blocks

File: controllers/visitor.js (hunk 1 above)
Comment (haydarai) [reply controllers/visitor.js]: What do you mean? Isn't it's defined here? https://github.com/kuriakinzeng/logbook/blob/c07e0c6caac003e30c95f847e5497aa2f571e5c1/app.js line 134-137

Comment (dwiajik) [reply controllers/visitor.js]: Ah sorry, I thought there will be User role that cannot create Visitor. Turns out every role can create Visitor :bowing_man:

Comment (haydarai) [reply controllers/visitor.js]: So it's all good then?

File: models/Visitor.js (hunk 2 above)
Comment (haydarai) [reply models/Visitor.js]: I'll fix it later today, thanks for the heads up!

**Comment by kuriakinzeng** [comment]
I have resolved the conflict @haydarai 
Do you have data I can test the endpoints with?

**Comment by haydarai** [comment]
@kuriakinzeng 
```
{
  "name": "Haydar Ali Ismail",
  "dob": "1994-06-18T02:45:00.000Z",
  "nric": "1065266604",
  "membershipId": "1065266604",
  "membershipExpiry": "1994-06-18T02:45:00.000Z"
}
```

**Pull request merged**